ProcessMaker/processmaker
# Resolve Project Member Filter Issue and Improve Data Handling

## Issue & Reproduction Steps
This PR addresses the issue where the members filter did not populate with project members as expected.

## Solution
In this PR, we have implemented the following solutions:

- Established a Project Member relationship in both the User and Group models.
- Established User and Group relationships in the Project Member model.
- Refactored the data retrieval process from the controller.
- Improved the formatting of User and Group data.

## How to Test

1. Ensure you have projects with members created.
2. Navigate to the Project Designer page and select the 'Filter' button.
3. Check the Members dropdown to ensure that members now appear as expected.
4. Confirm that you can successfully filter by members.


## Related Tickets & Packages
- Ticket [FOUR-11012](https://processmaker.atlassian.net/browse/FOUR-11012)
-  [Projects Package PR](https://github.com/ProcessMaker/package-projects/pull/40)

ci:next
ci:package-projects:observation/FOUR-11012

## Code Review Checklist


[FOUR-11012]: https://processmaker.atlassian.net/browse/FOUR-11012?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
